### PR TITLE
feat(phone/photo): emit an event when exiting/entering photo mode

### LINKED ISSUE #384: Enhancement: Event for hiding ui components on picture taken
### Application

Camera

### Dependency

No

### Feature Description

Essentially a net event that is triggered when hud components should be hid. Like the `ClearHelp(bool);` function we have where the ui components are cleared before the "picture" is taken.

Would resolve issues such as peoples huds/maps being on pictures and would be super simple to add docs for. 

image

Would end up being something like 2 events. One for show and one for hide or we can just use a bool instead of 2 events.

Below is a lua example for how server owners could utilize it:
```lua
-- 2 Events
AddEventHandler('npwd:hideHudComponents', function ()
    DisplayRadar(false) -- Hides map
    -- Add hud here
end)

AddEventHandler('npwd:showHudComponents', function ()
    DisplayRadar(true) -- Shows map
     -- Add hud here
end)

-- 1 Event but use a bool for showing components
AddEventHandler('npwd:showHudComponents', function (visible)
    if visible then
        DisplayRadar(true) -- Show map
        -- show hud
    else
        DisplayRadar(false) -- hide map
        -- hide hud
    end
end)
```
This would be the best solution imo as server owners can fix their own problems without rebuilding phone.

### PR DESCRIPTION
**Pull request by TasoOneAsia**
**Pull Request Description**

Implements and closes #384 

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

File: resources/client/cl_photo.ts [hunk 1]
```diff
@@ -43,11 +43,16 @@ RegisterNuiCB<void>(PhotoEvents.TAKE_PHOTO, async (_, cb) => {
...
-
   SetNuiFocus(false, false);
 
   inCameraMode = true;
 
+  // We want to emit this event for UI handling in other resources
+  emit(PhotoEvents.NPWD_PHOTO_MODE_STARTED);
+
+  // Player shouldnt have radar on screen
+  DisplayRadar(false);
```
Comment (RockySouthpaw) [inline resources/client/cl_photo.ts:54]: You shoukd check if the radar is already disabled before disabling it and then assign a bool here for later use if npwd disables it.

“why rocko?”
In the event a server has no minimap, we don’t want to enable it later. This would work as the initial check to see if they had a minimap to begin with. 

You can use `IsRadarHidden()` to check. It will return false if it’s showing.

File: resources/client/cl_photo.ts [hunk 2]
```diff
@@ -68,10 +73,14 @@ RegisterNuiCB<void>(PhotoEvents.TAKE_PHOTO, async (_, cb) => {
     }
     displayHelperText();
   }
+
   ClearHelp(true);
+  // We can now signal to other resources for ending photo mode
+  emit(PhotoEvents.NPWD_PHOTO_MODE_ENDED);
+  DisplayRadar(true);
```
Comment (RockySouthpaw) [inline resources/client/cl_photo.ts:80]: Same here

**Review by RockySouthpaw — CHANGES_REQUESTED** [review]

**Comment by TasoOneAsia** [comment]
Actually thinking about this PR and your comments more, I feel that it would be best to leave this to third party scripts through just the event. Otherwise, no matter what, edge cases may appear regarding HUD management.

**Comment by RockySouthpaw** [comment]
> Actually thinking about this PR and your comments more, I feel that it would be best to leave this to third party scripts through just the event. Otherwise, no matter what, edge cases may appear regarding HUD management.

Sg2m, will test shortly

**Comment by RockySouthpaw** [comment]
Would it be bad to just pass a bool instead of using 2 events? 

Idk if you can help it but waiting on image uploading before showing hud components may not be the best idea imo as image uploading can get hung then the components will never show. 

Otherwise, this works well in testing. 

```lua
AddEventHandler('npwd:PhotoModeStarted', function()
    LocalPlayer.state:set('pictureMode', true, true)
    DisplayRadar(false)
    RemoveMultiplayerHudCash()
    RemoveMultiplayerBankCash()
    TriggerEvent('WH:hideHud')
end)

AddEventHandler('npwd:PhotoModeEnded', function ()
    LocalPlayer.state:set('pictureMode', false, true)
    if not Config.hideRadar then DisplayRadar(true) end
    TriggerEvent('WH:showHud')
end)
```

https://i.imgur.com/bxqTwIR.jpg

**Review by RockySouthpaw — APPROVED** [review]
